backend/src/main/java/com/github/NicolaiKopka/web_page_services/main_page/MainPageService.java

```
@@ -22,7 +24,49 @@ public List<Movie> getPopularMovies() {
         return movieDBApiConnect.getPopularMovies();
     }
 
-    public void getSoundtrackOnSpotify() throws OAuthProblemException, OAuthSystemException {
-        spotifyApiConnect.getSpotifyStatusForMovie();
+    public StreamingStatusDTO getSoundtrackOnSpotify(String movieName) throws OAuthProblemException, OAuthSystemException {
+        StreamingStatusDTO streamingStatusDTO = new StreamingStatusDTO();
+        streamingStatusDTO.setMovieName(movieName);
+
+        try {
+            List<SpotifyAlbum> albums = spotifyApiConnect.getSpotifyListOfMovieAlbums(movieName);
+
+            SpotifyAlbum album = checkForExactSoundtrack(albums, movieName);
+            String link = album.getExternalURLs().getAlbumUrl();
+            streamingStatusDTO.getStreamingServiceStatus().put("spotify", true);
+            streamingStatusDTO.getAlbumLinks().put("spotify", link);
+
+        } catch (Exception e) {
```

I wouldn't catch all exceptions. That way bugs in the implementation, for example no null check, are hidden.